mlflow/models/utils.py

```diff
@@ -1621,3 +1621,16 @@ def _load_model_code_path(code_path: str, config: Optional[Union[str, Dict[str,
             ) from e
 
     return mlflow.models.model.__mlflow_model__
+
+
+def _flatten_nested_params(
+    d: Dict[str, Any], parent_key: str = "", sep: str = "/"
+) -> Dict[str, str]:
+    items: Dict[str, Any] = {}
+    for k, v in d.items():
+        new_key = f"{parent_key}{sep}{k}" if parent_key else k
+        if isinstance(v, dict):
+            items.update(_flatten_nested_params(v, new_key, sep=sep))
+        else:
```

Comment:
`v` could also be a list right? ie. this is valid yaml 
```
colors:
  - Red
  - Green
  - Blue
```

can we add a test for how this would be handled and confirm it is as expected?

Reply:
The else case should handle that. For dict we would like to flatten but for arrays, we keep it as it is. Also I have a test case for that in `tests/models/test_utils.py`.
I added another one for langchain and yaml with array as well so we can test it end to end :)